snow/engine/snowman/transitive_test.go

```diff
@@ -2762,54 +2767,39 @@ func TestEngineBubbleVotesThroughInvalidBlock(t *testing.T) {
 	reqID := new(uint32)
 	sender.SendGetF = func(_ context.Context, inVdr ids.NodeID, requestID uint32, blkID ids.ID) {
 		*reqID = requestID
-		if *asked {
-			t.Fatalf("Asked multiple times")
-		}
-		if blkID != blk1.ID() {
-			t.Fatalf("Expected engine to request blk1")
-		}
-		if inVdr != vdr {
-			t.Fatalf("Expected engine to request blk2 from vdr")
-		}
+		require.False(*asked)
+		require.Equal(blk1.ID(), blkID)
+		require.Equal(vdr, inVdr)
 		*asked = true
 	}
-	// Receive Gossip message for [blk2] first and expect the sender to issue a Get request for
-	// its ancestor: [blk1].
-	if err := te.Put(context.Background(), vdr, constants.GossipMsgRequestID, blk2.Bytes()); err != nil {
-		t.Fatal(err)
-	}
-
-	if !*asked {
-		t.Fatalf("Didn't ask for missing blk1")
-	}
+	// this engine receives Gossip message for [blk2] which is "unknown" in this engine
+	// thus sending a Get request for its ancestor [blk1]
+	// (see above for expected "Get" request)
```

Comment:
This ends up reading as if we will always request the ancestor of an unknown block. We request unknown ancestors of blocks when attempting to issue the dependent block
```suggestion
	// This engine receives a Gossip message for [blk2] which was "unknown" in this engine.
	// The engine thus learns about its ancestor [blk1] and should send a Get request for it.
	// (see above for expected "Get" request)
```